client/src/client_proxy.rs

```diff
@@ -519,6 +551,35 @@ impl ClientProxy {
         Ok(Some(file))
     }
 
+    /// Submit a transaction to the network given the unsigned raw transaction, sender public key
+    /// and signature
+    pub fn submit_signed_transaction(
+        &mut self,
+        raw_txn: RawTransaction,
+        public_key: Ed25519PublicKey,
+        signature: Ed25519Signature,
+    ) -> Result<AddressAndSequence> {
```

Comment:
Why do you need to return the Address here since it can be derived from the public key input?

Reply:
Actually, help me understand why do you need a return value at all? Both the address and sequence number can be derived from the inputs already upon successful return of the function.

Reply:
You are right that these values can be derived from the inputs.
However:
1. Returning `AddressAndSequence` saves the need of the caller to compute the address himself (convenience) and in order for him to query the sequence number he would need to do another communication call to the node given the sender's address, while this is already calculated as part of the submission (performance and convenience).
It is very likely that the caller will be interested in these two values, because together they consist the "transaction id" which has just been sent.
2. I tried to be aligned with the return value of `transfer_coins` which returns `IndexAndSequence`. The only difference here is that instead of index we return an address, because an index means that the account is necessarily a part of the current wallet, whereas in our case this is not true (the sender's key pair can be external to the wallet).

Reply:
So, sender address already exists as part of the raw transaction as input to this function. When calling this function you'd have had the sender address already. 
The sequence number is also part of the raw transaction, and in fact if you look at your own code you return exactly that for the happy case. 
My feeling is that it's a bit confusing for a function to return part of its inputs as its return values.

Reply:
You're right!
I dropped the `AddressAndSequence` struct and now the function returns just a `Result<()>`.